forms/TreeDropdownField.php

```diff
@@ -460,15 +460,26 @@ protected function populateIDs() {
 				if ($row->ParentID) $parents[$row->ParentID] = true;
 				$this->searchIds[$row->ID] = true;
 			}
+			
+			$sourceObject = $this->sourceObject;
+			$qSuffix = '';
+			// The parent of a file can also be a file (and not necessarily a folder, see File::has_one()) but we are 
+			// only interested in displaying folders in the dropdown field, when searching for folders.
+			if ($sourceObject == 'Folder') {
+				$qSuffix .= '"ClassName" IN (\''.implode('\',\'',Convert::raw2sql(ClassInfo::subclassesFor('Folder'))).'\')';
```

Comment:
True this is not needed.
When fixing the issue I had some files returned as part of the list of parent folders, but I can't reproduce it now. Maybe I added the check against ClassName before I turned the query from raw to ORM based.